src/sinks/prometheus/remote_write.rs

```diff
@@ -127,16 +140,45 @@ impl SinkConfig for RemoteWriteConfig {
 
         let client = HttpClient::new(tls_settings, cx.proxy())?;
         let tenant_id = self.tenant_id.clone();
-        let auth = self.auth.clone();
+        let http_auth = match &self.auth {
+            Some(PrometheusRemoteWriteAuth::Basic { user, password }) => Some(Auth::Basic {
+                user: user.clone(),
+                password: password.clone(),
+            }),
+            _ => None,
+        };
+
+        let (credentials_provider, aws_region) = match &self.auth {
+            Some(PrometheusRemoteWriteAuth::Aws(aws_auth)) => {
+                let region = self
+                    .aws
+                    .as_ref()
+                    .map(|config| config.region())
+                    .ok_or(Errors::AwsRegionRequired)?
+                    .ok_or(Errors::AwsRegionRequired)?;
+
+                (
+                    Some(aws_auth.credentials_provider(region.clone()).await?),
+                    Some(region),
+                )
+            }
+            Some(PrometheusRemoteWriteAuth::Basic { .. }) | None => (None, None),
+        };
```

Comment:
Does it make sense to merge these into one `match`? ie:
```rust
    let (http_auth, credentials_provider, aws_region) = match &self.auth …
```

Reply:
It almost makes too much sense 😄